### PR TITLE
Updating Sample Data set

### PR DESCRIPTION
**Pull request by ehsteve**
This PR updates the sample data set to the use the new sample data. Also provides a gallery example which provides an overview of this new data set.

### CODE REVIEW

File: examples/2011_06_07_sampledata_overview.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+plt.show()
+
+###############################################################################
+# Let's take a look at the GOES XRS data.
+goes = sunpy.timeseries.TimeSeries(sample_data.GOES_XRS_TIMESERIES)
+goes.peek()
+plt.show()
+
+###############################################################################
+# Next let's investigate the AIA full disk images that are available. FYI,
```
Comment (Cadair) [inline examples/2011_06_07_sampledata_overview.py]: 'FYI'... Really?! ;)

File: examples/2011_06_07_sampledata_overview.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+plt.show()
+
+###############################################################################
+# As well as RHESSI.
+rhessi_map = sunpy.map.Map(sample_data.RHESSI_IMAGE)
+fig = plt.figure()
+#ax = plt.subplot(projection=rhessi_map)
+ax = plt.subplot()
+rhessi_map.plot()
+plt.show()
```
Comment (Cadair) [inline examples/2011_06_07_sampledata_overview.py]: Might be cool to show the flare in goes, and at least mention the other data?

Comment (ehsteve) [inline examples/2011_06_07_sampledata_overview.py]: I show the flare in GOES on line 38.

Comment (Cadair) [inline examples/2011_06_07_sampledata_overview.py]: so you do

**Comment by ehsteve** [comment]
ping @wafels @dpshelio @ayshih. I need more reviews!

**Comment by Cadair** [comment]
@ehsteve I rebased this for you.

**Comment by Cadair** [comment]
again bite me HEK.

**Pull request merged**